src/Raven.Server/Indexing/VoronIndexInput.cs

```diff
@@ -137,21 +113,27 @@ public override long FilePointer
         {
             get
             {
-                AssertNotDisposed();
-
+                ThrowIfDisposed();
+                ThrowIfCancellationRequested();
                 return _stream.Position;
             }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void AssertNotDisposed()
+        private void ThrowIfDisposed()
         {
-            if (_cts.IsCancellationRequested)
-                throw new ObjectDisposedException("VoronIndexInput");
             if(_currentTransaction.Value == null)
                 throw new ObjectDisposedException("No Transaction in thread");
             if (_currentTransaction.Value.LowLevelTransaction.IsDisposed)
                 throw new ObjectDisposedException("No Transaction in thread");
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ThrowIfCancellationRequested()
+        {
+            if (_cts.IsCancellationRequested)
+                throw new OperationCanceledException("VoronIndexInput");
```

Comment:
The idea is to remove the `throw` statement and all the code it generates from the hot codepath.
What you did is just moved both `if` the and the `throw` to a method, so we'll have to force to go over it anyway.
Just the `throw` should be in its own method.